src/main/java/com/conveyal/datatools/manager/jobs/DeployJob.java

```diff
@@ -655,6 +671,21 @@ private void replaceEC2Servers() {
                     }
                 }
             }
+            // Wait for a recreate graph build job to complete if one was started
+            if (recreateBuildImageExecutor != null) {
+                status.update("Waiting for recreate graph building image job to complete", 95);
+                while (!recreateBuildImageJob.status.completed) {
+                    try {
+                        // wait 1 second
+                        Thread.sleep(1000);
+                    } catch (InterruptedException e) {
+                        recreateBuildImageExecutor.shutdown();
+                        status.fail("An error occurred while waiting for the graph build image to be recreated", e);
```

Comment:
Should an exception here constitute a failure of the entire deploy job? Also does `recreateBuildImageJob` need to be failed in this catch block?

Reply:
Yes, I think it should. My logic is that if one part of the DeployJob fails (including this recreate build image job), then the entire job should be failed. There should be enough feedback via the other jobs to show the user that just the recreate build image job failed if that was the only failure.

I added some logic to also fail the build image job just in case.